### PR TITLE
Parameters: handle Parameters as dependencies

### PR DESCRIPTION
**Pull request by kmantel**
Correct bug where a Parameter dependency would fail to be respected if a Parameter P specified its dependency on Parameter Q as a Parameter and not the string 'Q', and Q had a dependency of its own

### CODE REVIEW

File: psyneulink/core/globals/parameters.py [hunk 1]
```diff
@@ -2124,9 +2124,18 @@ def __init__(self, owner, parent=None):
         for alias_name in aliases_to_create:
             setattr(self, alias_name, ParameterAlias(name=alias_name, source=getattr(self, alias_name).source))
 
-        for param, value in self.values(show_all=True).items():
+        values = self.values(show_all=True)
+        for param, value in values.items():
             self._validate(param, value.default_value)
 
+            if value.dependencies is not None:
```
Comment (jvesely) [inline psyneulink/core/globals/parameters.py:2131]: just a drive-by comment:
Can this be simplified?
```
value.dependencies = {getattr(d, 'name', d) for d in value.dependencies}
assert all(dep in values for dep in value.dependencies)
```

Comment (kmantel) [reply psyneulink/core/globals/parameters.py:2131]: Yes, thanks. I always forget about the default option for getattr.
